globals.d.ts

```diff
@@ -52,6 +52,137 @@ export interface DatabaseObject {
   [key: string]: unknown;
 }
 
+/**
+ * Search result object; similar to `DatabaseObject` but without some properties.
+ */
+export type SearchResultsObject =
+  | SearchResultsObjectProps
+  | SearchResultsObjectGenerics;
+
+interface SearchResultsObjectProps {
+  "@id": string;
+  "@type": string[];
+}
+
+interface SearchResultsObjectGenerics {
+  [key: string]: unknown;
+}
+
+/**
+ * The `properties` object of a schema.
+ */
+export type SchemaProperties = SchemaPropertiesProps | SchemaPropertiesGenerics;
+
+export interface SchemaPropertiesProps {
+  title: string;
+  type: string;
+}
+
+export interface SchemaPropertiesGenerics {
```

Comment:
If we we used known json-schema types, or copied them, I don't think we'd have to have a generic key: unknown since we'd know the types of properties and what values they take and we'd have to cast less I think. But that might be overkill as well.